tools/check-regions-drift.py

```
@@ -0,0 +1,261 @@
+                raise CheckError(f"<string name=\"{REGIONS_URL_RESOURCE_NAME}\"> in {resource_path} is empty")
+            return url
+    raise CheckError(f"no <string name=\"{REGIONS_URL_RESOURCE_NAME}\"> in {resource_path}")
+
+
+def fetch(url):
+    """Response body at `url` as text; an unreachable directory is a CheckError, not drift."""
+    try:
+        # An explicit User-Agent is required, not politeness: the directory host answers 403 to
+        # urllib's default `Python-urllib/3.x`. Built inside the try because a malformed URL fails
+        # here, at Request(), rather than at urlopen().
+        request = urllib.request.Request(url, headers={"User-Agent": USER_AGENT})
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)